composer/core/callback.py

```diff
@@ -200,6 +200,11 @@ def after_train_batch(self, state: State, logger: Logger) -> None:
     def batch_end(self, state: State, logger: Logger) -> None:
         """Called on the :attr:`~Event.BATCH_END` event.
 
+        .. note::
+
+            The :attr:`state.timer.batch` is incremented between `:attr:`~Event.BATCH_START` and
+            :attr:`~Event.BATCH_END`.
```

Comment:
Maybe, incremented just before `~Event.BATCH_END`? Makes it a bit more clear when it is incremented.